tinygrad/engine/schedule.py

```diff
@@ -338,9 +341,25 @@ def _as_const(u:UOp, val:ConstType) -> UOp:
   st = (base:=ShapeTracker.from_shape(())).reshape((1,)*len(u.shape)).expand(u.shape)
   return UOp(Ops.VIEW, u.dtype, (u.buf_uop, UOp.const(u.dtype, val)), base).view(st)
 
+def simplify_const_reduce(ctx, src:UOp, to_store:UOp, **kwargs) -> Optional[UOp]:
+  # must be unmasked
+  if any(v.mask is not None for v in unwrap(src.st).views): return None
+  prshape = prod(unwrap(to_store.src[0].st).shape[i] for i in to_store.arg[1])
+  val = src.src[1].arg
+  match to_store.arg[0]:
+    case Ops.ADD: ret = val*prshape
+    case Ops.MUL: ret = val**prshape
+    case _: ret = val
```

Comment:
Careful, I think you should add Ops.MAX here and assert (or return None) if it's something unhandled